app/Http/Controllers/PollerGroupController.php

```diff
@@ -0,0 +1,189 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Device;
+use App\Models\Poller;
+use App\Models\PollerGroups;
+use App\Models\PollerCluster;
+use Illuminate\Validation\Rule;
+use Illuminate\Http\Request;
+use LibreNMS\Alerting\QueryBuilderFilter;
+use LibreNMS\Alerting\QueryBuilderFluentParser;
+use LibreNMS\Config;
+use Toastr;
+
+class PollerGroupController extends Controller
+{
+    public $rrdstep;
+    public $isDistributedPoller;
+    public $defaultPollerId;
+
+    public $defaultPollerTab = 'poller';
+
+    public $defaultGroup = ['id' => 0,
+                             'group_name' => 'General',
+                             'descr' => ''];
+    public $defaultPollerMarker = '(default Poller)';
+
+    public function __construct()
+    {
+        $this->authorizeResource(PollerGroup::class, 'poller_groups');
+
+        $this->rrdstep = \LibreNMS\Config::get('rrd.step');
+
+        $this->isDistributedPoller = \LibreNMS\Config::get('distributed_poller');
+        $this->defaultPollerId = \LibreNMS\Config::get('distributed_poller_group');
+    }
+
+    public function index(Request $request)
```

Comment:
Maybe we should split this into one function per tab, then it's easier to use the built in laravel route features etc.

Reply:
also, if we change the urls to `poller-groups/log`, then we can use `route('poller-groups.log')` for links etc.

Reply:
ok, i'll change it

Reply:
@Jellyfrog 
i'm not such deep in Laravel (at moment), what do i have to change for getting it to worked with
poller-groups/poller
poller-groups/groups
poller-groups/performance
poller-groups/log
?

Reply:
Just split it into one Controller per tab... it gets rid of a lot of if/else switch statements.

Reply:
@murrant 
sounds a little bit redundant
every tab as seperate function like @Jellyfrog sounds more useable

Reply:
Spontaneously i would use multiple get() to define routes

Reply:
updates to murrants idea `Route::get('poller-groups/{tab}', 'PollerGroupController@index');`